ethclient/simulated/backend.go

```diff
@@ -142,12 +143,16 @@ func (n *Backend) Close() error {
 		n.client.Close()
 		n.client = simClient{}
 	}
+	var err error
 	if n.beacon != nil {
-		err := n.beacon.Stop()
+		err = n.beacon.Stop()
 		n.beacon = nil
-		return err
 	}
-	return nil
+	if n.node != nil {
+		err = errors.Join(n.node.Close())
```

Comment:
err = errors.Join(err, n.node.Close())

It's possible the err was non-nil, isn't it?